### PR TITLE
Make rocblas_gemm_flags_fp16_alt_impl in MHA and MLP backward compatible with old PyTorch versions

### PR DESCRIPTION
**Pull request by hubertlu-tw**

### CODE REVIEW

**Review by jithunnair-amd — CHANGES_REQUESTED** [review]
@hubertlu-tw Why do the MHA files not use the ROCM_BACKWARD_PASS_GUARD define?

**Comment by hubertlu-tw** [comment]
> @hubertlu-tw Why do the MHA files not use the ROCM_BACKWARD_PASS_GUARD define?

It should use the ROCM_BACKWARD_PASS_GUARD define in MHA, too. Thanks for pointing out.

File: apex/contrib/csrc/multihead_attn/encdec_multihead_attn_cuda.cu [hunk 1]
```diff
@@ -321,9 +318,16 @@ std::vector<torch::Tensor> bwd_cuda(
...
   //TORCH_CUDABLAS_CHECK(cublasSetMathMode(handle, CUBLAS_TENSOR_OP_MATH));
-  #if USE_GEMM_FLAGS_FP16_ALT_IMPL
-    flags = at::BackwardPassGuard::is_backward_pass() ? rocblas_gemm_flags_fp16_alt_impl : 0;
-  #endif 
+  #ifdef __HIP_PLATFORM_HCC__
+    #define PYTORCH_ROCBLAS_VERSION_DECIMAL (ROCBLAS_VERSION_MAJOR * 100 + ROCBLAS_VERSION_MINOR)
+    #define USE_GEMM_FLAGS_FP16_ALT_IMPL (PYTORCH_ROCBLAS_VERSION_DECIMAL >= 242)
+    #if USE_GEMM_FLAGS_FP16_ALT_IMPL
+      #ifdef ROCM_BACKWARD_PASS_GUARD
+        flags = at::BackwardPassGuard::is_backward_pass() ? rocblas_gemm_flags_fp16_alt_impl : 0;
```
Comment (jithunnair-amd) [inline apex/contrib/csrc/multihead_attn/encdec_multihead_attn_cuda.cu:326]: This is inside 3 layers of #if conditions, where's the fallback?

Comment (hubertlu-tw) [reply apex/contrib/csrc/multihead_attn/encdec_multihead_attn_cuda.cu:326]: The variable (default value = 0) defined [here](https://github.com/ROCmSoftwarePlatform/apex/blob/dev/hubertlu/rocblas_backward_compatible/apex/contrib/csrc/multihead_attn/strided_batched_gemm.h#L27) is used in the MHA CUDA files.

**Pull request merged**